### DISCUSSION TITLE
Reducing CPU usage on Intel Xeon server

### DISCUSSION THREAD

**Discussion by YBonline** [blakeblackshear/frigate#5436] [Ask A Question]
I have a Dell R440 server that has Dual Xeon Silver 4208 CPUs, 64GB of RAM, along with a Google Coral for object detection, which is working great. I currently have 20 cameras. Frigate is running fine (the server also runs a bunch of other applications for me as well, although Frigate seems to be one of the largest consumer of resources).

I'm planning to increase maybe up to 30 cameras soon, and have noticed that my CPU usage seems to be averaging around 75-80% right now, so I was curious to make sure I'm not fully loading my server, what my best approach was?

My understanding is the Xeons don't support QSV or any sort of hardware acceleration, but I am not transcoding any video. One thing I am confused about is, is the creation of the snapshots for Coral to process benefit from a GPU or hardware acceleration? Is my best bet to reduce CPU usage to get a graphics card? If so, which one is best? A Quadro card?

I understand its frequently recommended to get a normal modern i7/i9 to save on power, but I run a fair amount of stuff on this server, and if I went that route, I'd likely have to run it in addition to the R440, so I don't think it'd make sense to run an additional computer/server for Frigate with a separate CPU? Also is that advice for people running a normal number of cameras or is it still appropriate for running 30 cameras?

**Comment by blakeblackshear** [comment]
Frigate has to decode the video streams to perform motion detection and send the relevant portions of the frames to the Coral for object detection. The Coral doesn't help decode video streams at all, so this is where hardware acceleration helps. They are completely different things. Most of your load is probably the ffmpeg processes decoding the video streams.

So yes, a GPU that can take the load for decoding video streams is your best option.

**Reply by YBonline** [reply]
Thanks! Is there a certain video card or driver that works better then others? What is most popular? The server only has room for a single slot card

**Comment by YBonline** [comment]
So I went ahead and bought a NVIDIA T1000 and it definetely seemed to reduce my load, but I seem to be running out of memory on the GPU:

With 20 cameras, I seem to have 2 that usually are failing to load, and the culprit appears to be:

 [AVHWDeviceContext @ 0x560afe2bbe00] cu->cuCtxCreate(&hwctx->cuda_ctx, desired_flags, hwctx->internal->cuda_device) failed -> CUDA_ERROR_OUT_OF_MEMORY: out of memory

It has 4GB of RAM on the GPU. Is there anything that can be done to fix this or share system memory? Or do I need to return this and try to get the 8GB model?

**Reply by NickM-27** [reply]
If your GPU don't have enough memory I don't think there's much that can be done. Maybe could tweak ffmpeg to reduce GPU memory but I've never looked into that
